content/en/blog/falcosidekick-knative.md

```diff
@@ -0,0 +1,460 @@
+```
+
+This will install a
+[Knative Service](https://knative.dev/docs/serving/#serving-resources) that will
+consume the Falco events sent by falcosidekick (to the broker), some
+[RBAC](https://github.com/n3wscott/falco-drop/blob/v0.1.0/config/rbac.yaml) to
+enable that service to delete pods, and a Knative Trigger to register this
+consumer for events from the
+[Knative Eventing Broker](https://knative.dev/docs/eventing/broker/).
+
+#### Consumer KService
+
+The simplified go code in use is like the following:
+
+```go
+func main() {
+	...setup context...
+	kc := kubeclient.Get(ctx)
+
+	// Make a CloudEvents Client.
+	c, _ := cloudevents.NewDefaultClient(p)
+
+	// StartReceiver is blocking, it will deliver events to the inline function.
+	c.StartReceiver(ctx, func(event cloudevents.Event) {
+		// Filter based on source and type.
+		if event.Source() == "falco.org" && event.Type() == "falco.rule.output.v1" {
+
+			// Extract the Falco event Payload
+			payload := &FalcoPayload{}
+			if err := event.DataAs(payload); err != nil {
+				return
+			}
+
+			// Only react to "Terminal shell in container" triggered rules.
+			if payload.Rule == "Terminal shell in container" {
+				if err := kc.CoreV1().Pods(payload.Fields.Namespace).Delete(ctx, payload.Fields.Pod, metav1.DeleteOptions{}); err != nil {
+					log.Println("failed to delete pod from event:", err)
+					return
+				}
+				log.Printf("[%s] deleted %s from %s because %s\n", payload.Rule, payload.Fields.Pod, payload.Fields.Namespace, payload.Output)
+			}
+		}
+	})
+}
+```
+
+The full implementation can be found in the
+[falco-drop](https://github.com/n3wscott/falco-drop/blob/main/cmd/drop/main.go)
+repo.
+
+> Pro-tip: if you are developing in Go for Kubernetes, take a look at
+> [ko](https://github.com/google/ko). `ko` enables containerizing go
+> applications without needing a Dockerfile.
+
+Even though the Trigger only delivers events that match the Trigger filter, it
+is a good idea to validate the event that the function is receiving, which is
+why we are validating again in the above code (trust, but verify).
+
+#### Eventing Triggers
+
+The Trigger configures the Broker for a subscriber to be invoked when the Broker
+ingresses an event that matches the `spec.filter` settings.
+
+```yaml
+apiVersion: eventing.knative.dev/v1
+kind: Trigger
+metadata:
+  name: drop
+  namespace: default
+spec:
+  broker: default
+  filter:
+    attributes:
+      source: falco.org
+      type: falco.rule.output.v1
+  subscriber:
+    ref:
+      apiVersion: serving.knative.dev/v1
+      kind: Service
+      name: drop
+```
+
+Here we are requesting that the broker only deliver events that have the
+attributes (CloudEvent attributes) of `source=falco.org` and
+`type=falco.rule.output.v1`. These events are delivered to our subscriber
+KService.
+
+![Eventing Topology](https://github.com/n3wscott/falco-drop/raw/main/img/graph.png)
```

Comment:
This is awesome. Might be better if you could drop the sockeye bit to reduce cognitive load